tests/skip.rs

```diff
@@ -0,0 +1,61 @@
+#[macro_use]
+extern crate parity_codec_derive;
+
+use parity_codec::{Encode, Decode};
+
+#[test]
+fn enum_struct_test() {
+	#[derive(PartialEq, Debug, Default)]
+	struct UncodecType;
+
+	#[derive(PartialEq, Debug)]
+	struct UncodecUndefaultType;
+
+	#[derive(PartialEq, Debug, Encode, Decode)]
+	enum Enum<T=UncodecType, S=UncodecUndefaultType> {
+		#[codec(skip)]
+		A(S),
```

Comment:
Shouldn't this fail to compile when you want to decode it? As `S` is not insatiable?

Reply:
I don't think so, isn't that our usecase with enum Event ? we want some variant to be skipped in the encoded representation imo

Reply:
I don't want it to fail, but as `S` does not implement `Default`, `decode` should not be callable?